lib/Runtime/PlatformAgnostic/Platform/Common/UnicodeText.Common.cpp

```diff
@@ -88,14 +98,18 @@ namespace PlatformAgnostic
             ///     Otherwise, they're the same to continue scanning
             ///  else they're both not numbers:
             ///     Compare lexically till we find a number
+            ///  if either of the strings current character is a null character continue scanning
             /// The algorithm treats the characters in a case-insensitive manner
             ///
-            int LogicalStringCompareImpl(const char16* p1, const char16* p2)
+            int LogicalStringCompareImpl(const char16* p1, int p1size, const char16* p2, int p2size)
             {
                 Assert(p1 != nullptr);
                 Assert(p2 != nullptr);
 
-                while (*p1 && *p2)
+                int str1size = 0;
```

Comment:
I find this name slightly confusing, since I was expecting that the size parameter would be fixed. It looks like it is only used to make sure we don't walk past the end of the strings, would it make sense to instead compute something like `const char16* str1End = p1 + p1size, *str2End = p2 + p2Size` and then just check `p1 < str1End && p2 < str2End` in the loop condition? That way we also don't accidentally bump either pointer without bumping the length-so-far.

Reply:
Completely agree with you. Thanks!